### PR TITLE
update revapi of spring package for feature 4.0

### PR DESCRIPTION
**Pull request by Netyyyy**
Remove suppression of spring package and add targeted suppression.

### CODE REVIEW

**Review by weidongxu-microsoft — APPROVED** [review]
OK for AllocationPolicyTests.java

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json [hunk 1]
```diff
@@ -105,6 +101,72 @@
     "configuration": {
       "ignore": true,
       "differences": [
+        {
+          "regex": true,
+          "code": "java.class.externalClassExposedInAPI",
+          "new": "(interface|enum) org.junit.jupiter.api..*",
+          "justification": "junit.jupiter classes are allowed to be exposed."
+        },
```
Comment (saragluna) [inline eng/code-quality-reports/src/main/resources/revapi/revapi.json:109]: Why do we add this? We shouldn't expose any testing classes in our API.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json [hunk 2]
```diff
@@ -105,6 +113,54 @@
...
+          "code": "java.class.externalClassExposedInAPI",
+          "new": "class com.azure.data.appconfiguration..*",
+          "justification": "azure.data.appconfiguration classes are allowed to be exposed."
+        },
+        {
+          "regex": true,
+          "code": "java.class.externalClassExposedInAPI",
+          "new": "class com.azure.identity..*",
+          "justification": "azure.identity classes are allowed to be exposed."
+        },
```
Comment (saragluna) [inline eng/code-quality-reports/src/main/resources/revapi/revapi.json:151]: Can we add this to this rule? 

```java
{
          "regex": true,
          "code": "java\\.class\\.externalClassExposedInAPI",
          "new": "(interface|class|enum) com\\.azure\\.(communication\\.common|core|cosmos|data\\.schemaregistry|messaging\\.eventgrid|resourcemanager|security\\.keyvault|storage).*",
          "justification": "SDK classes are allowed to be exposed by dependencies using them."
        },
```

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json [hunk 3]
```diff
@@ -105,6 +113,54 @@
...
+          "code": "java.class.externalClassExposedInAPI",
+          "new": "(class|interface|enum|method) com.azure.spring..*",
+          "justification": "azure.spring classes are allowed to be exposed."
+        },
+        {
+          "regex": true,
+          "code": "java.class.nonPublicPartOfAPI",
+          "new": "class com.azure.spring.messaging.container.DefaultMessageListenerContainer",
+          "justification": "We allow this in azure.spring.messaging."
+        },
```
Comment (saragluna) [inline eng/code-quality-reports/src/main/resources/revapi/revapi.json:163]: Why do we need this?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json (hunk 3 above)
Comment (saragluna) [reply eng/code-quality-reports/src/main/resources/revapi/revapi.json]: Okay, I figured it out. The visibility of `DefaultMessageListenerContainer` is package level but it's returned from the `AbstractAzureListenerContainerFactory.createListenerContainer` method.

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**